### DISCUSSION TITLE
Flat profile well simulation

### DISCUSSION THREAD

**Discussion by MehtaMeet54** [GazzolaLab/PyElastica#104] [Q&A]
For simulating a well/tube/tunnel I made an interaction class and for response comparison I was attempting to benchmark against the InteractionPlane and the AnisotropicFrictionalPlane and the models behave well but not similarly, as in for some range of parameters the AnisotropicFrictionalPlane does work better and settle down in a known time period and my model blows up to Nan and for some the AnisotropicFrictionalPlane just keeps on oscillating and my model settles down with small oscillations. I have attached the work bellow.

Are there any ways to resolve these inconsistencies and any way to improve these interaction class?

[Flat bed simulations.zip](https://github.com/GazzolaLab/PyElastica/files/8839749/Flat.bed.simulations.zip)

**Comment by armantekinalp** [comment]
Hi @MehtaMeet54 

Is the InteractionPlane you mentioned your model or [interaction plane](https://github.com/GazzolaLab/PyElastica/blob/master/elastica/interaction.py#L135) in Elastica? 

Could you please provide more details? For example, some graphs, videos from simulations that are using the models you mentioned.

**Comment by MehtaMeet54** [comment]
Yes the interaction plane is the one that is present in Elastica.
Here is a video of the simulation.

https://user-images.githubusercontent.com/34564632/172357008-165ab670-979e-4bfe-b88c-ab95f64dd1f2.mp4

**Comment by armantekinalp** [comment]
Are you using `AnisotropicFrictionalPlane` class or `InteractionPlane`? 

You should use `AnisotropicFrictionalPlane` since you cannot add `InteractionPlane` into simulator. If you don't want friction and only plane response  just set `static_mu_array` and `kinetic_mu_array` to zero.

**Reply by MehtaMeet54** [reply]
I am using `AnisotropicFrictionalPlane` with `static_mu_array` and `kinetic_mu_array` to zero.
